File: test/support/r/naive_bayes.R
library(pmml, quiet=TRUE)
library(e1071)

data <- read.csv("test/support/mpg.csv")
model <- naiveBayes(drv ~ displ + year + cyl + class, data, laplace=1)

pmml <- toString(pmml(model, predicted_field="drv"))
write(pmml, file="test/support/r/naive_bayes.pmml")

print(predict(model, data[0:10,]))

File: test/support/python/naive_bayes.py
import pandas as pd
from sklearn.preprocessing import OneHotEncoder
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.naive_bayes import GaussianNB
from sklearn2pmml import sklearn2pmml
from sklearn2pmml.decoration import ContinuousDomain, CategoricalDomain
from sklearn2pmml.pipeline import PMMLPipeline
from sklearn2pmml.feature_extraction.text import Splitter
from sklearn_pandas import DataFrameMapper

data = pd.read_csv("test/support/mpg.csv")

numeric_features = ["displ", "year", "cyl"]
categorical_features = ["class"]
text_features = []

mapper = DataFrameMapper(
  [(numeric_features, [ContinuousDomain()])] +
  [([f], [CategoricalDomain(), OneHotEncoder()]) for f in categorical_features] +
  [(f, [CategoricalDomain(), CountVectorizer(tokenizer=Splitter())]) for f in text_features]
)

pipeline = PMMLPipeline([
  ("mapper", mapper),
  ("model", GaussianNB())
])
pipeline.fit(data, data["drv"])

sklearn2pmml(pipeline, "test/support/python/naive_bayes.pmml")

print(pipeline.predict(data[:10]))
